### PR TITLE
dep: update for release

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
part of https://github.com/pingcap/dm/issues/1337

### What is changed and how it works?
- [x] update parser https://github.com/pingcap/dm/pull/1356/commits/88f3fe3dbf77404d50373957946d28cd2779a133
- [x] update dumpling https://github.com/pingcap/dm/pull/1356/commits/1486fc24572a5739df49162223a86432bc0d4bfd
- [x] update tidb-tools (?) https://github.com/pingcap/dm/pull/1356/commits/47afa4e586c98000d6de691e9a96259f0f2ddb0e

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by lance6716** [comment]
Github action lint:

```
Post job cleanup.
/bin/tar --posix --use-compress-program zstd -T0 -cf cache.tzst -P -C /home/runner/work/dm/dm --files-from manifest.txt
Warning:  uploadChunk (start: 469762048, end: 503316479) failed: Cache service responded with 500
/home/runner/work/_actions/golangci/golangci-lint-action/v2/dist/post_run/index.js:4525
                        throw new Error(`Cache upload failed because file read failed with ${error.message}`);
                        ^

Error: Cache upload failed because file read failed with EBADF: bad file descriptor, read
    at ReadStream.<anonymous> (/home/runner/work/_actions/golangci/golangci-lint-action/v2/dist/post_run/index.js:4525:31)
    at ReadStream.emit (events.js:210:5)
    at internal/fs/streams.js:167:12
    at FSReqCallback.wrapper [as oncomplete] (fs.js:470:5)
```

**Comment by lance6716** [comment]
`online_ddl`

```
[2020-12-24T03:13:03.697Z] use sync_diff_inspector to check full dump data
[2020-12-24T03:13:03.698Z] check diff successfully
[2020-12-24T03:13:03.698Z] 1.1.0
[2020-12-24T03:13:11.860Z] 2020-12-24 11:13:10 ERROR Error 1205: Lock wait timeout exceeded; try restarting transaction
```

**Comment by lance6716** [comment]
`online_ddl_gh-ost` keeps failing

**Comment by lance6716** [comment]
waiting https://github.com/pingcap/dumpling/pull/222

**Comment by lance6716** [comment]
@lichunzhu please help verify, I think though the `metaconn` is not closed, `TransactionalConsistency` should take effects and unlock tables. Why doesn't it take effect?

**Comment by lance6716** [comment]
@lichunzhu PTAL

File: dumpling/util.go [hunk 1]
```diff
@@ -94,23 +77,7 @@ func parseExtraArgs(logger *log.Logger, dumpCfg *export.Config, args []string) e
...
-		ff, err2 := parseTableFilter(tablesList, filters)
-		if err2 != nil {
-			return err2
-		}
-		dumpCfg.TableFilter = ff // overwrite `block-allow-list`.
-		logger.Warn("overwrite `block-allow-list` by `tables-list` or `filter`")
-	}
-
-	return nil
+	return dumpCfg.ParseFromFlags(dumplingFlagSet)
```
Comment (lichunzhu) [inline dumpling/util.go:80]: Should this be after line `if noLocks {`?

**Review by lichunzhu — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
/run-all-tests

```
[2020-12-25T02:41:56.987Z] FAIL: server_test.go:1394: testMaster.TestOfflineMember
[2020-12-25T02:41:56.987Z] 
[2020-12-25T02:41:56.987Z] server_test.go:1458:
[2020-12-25T02:41:56.987Z]     c.Assert(err, check.IsNil)
[2020-12-25T02:41:56.987Z] ... value *status.statusError = &status.statusError{Code:2, Message:"[code=38043:class=dm-master:scope=internal:level=high], Message: master is not leader, and can't forward request to leader", Details:[]*any.Any(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0} ("rpc error: code = Unknown desc = [code=38043:class=dm-master:scope=internal:level=high], Message: master is not leader, and can't forward request to leader")
```

**Comment by lance6716** [comment]
PTAL @GMHDBJD

**Comment by GMHDBJD** [comment]
LGTM

**Pull request merged**